src/tests/suite/daos_test.h

```diff
@@ -306,7 +307,24 @@ async_overlap(void **state)
 static inline int
 test_case_teardown(void **state)
 {
-	assert_rc_equal(daos_event_priv_reset(true), 0);
+	sigset_t	sigset;
+	bool		force = false;
+
+	/*
+	 * If one of SIGFPE/SIGILL/SIGSEGV/SIGBUS/SIGSYS is in the signal mask, then the logic is
+	 * longjump from cmocka for handling the signal, then need force cleanup test environment.
+	 */
+	if (sigprocmask(0, NULL, &sigset) < 0) {
+		print_message("sigprocmask failure\n");
+	} else if (unlikely(sigismember(&sigset, SIGFPE) ||
+			    sigismember(&sigset, SIGILL) ||
+			    sigismember(&sigset, SIGSEGV) ||
+			    sigismember(&sigset, SIGBUS) ||
+			    sigismember(&sigset, SIGSYS))) {
+		force = true;
```

Comment:
it is good if it can check if it is longjump from cmocka internally, in that case it mostly like there are mem corruption already, maybe not worth to do the follow "cleanup".
How about just print fatal err log here first, to see if it will really hit this case in following test.
The change in event.c is not bad, but probably not must to do that now to avoid affect event internal handling. If will hit this failure later in CI tests, we can consider further later?

Reply:
It's good idea to print the error message when hit such trouble. As for whether force cleanup or not, I tend to do that. Because:

1. If force cleanup works well, then the subsequent tests can go ahead.
2. We may have not verified related cleanup logic under some fatal failure cases. We do that now, it may help to expose some potential bug(s).

Since the test has  already corrupted, the cleanup logic will not make the situation to be worse.